server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -161,9 +164,9 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
     final String defaultHost = hostFinder.getDefaultHost();
     request.setAttribute(HOST_ATTRIBUTE, defaultHost);
 
-    final boolean isQueryEndpoint = request.getRequestURI().startsWith("/druid/v2");
+    final String requestURI = request.getRequestURI();
 
-    if (isQueryEndpoint && HttpMethod.DELETE.is(request.getMethod())) {
+    if (requestURI.startsWith(QUERY_URL_V2) && HttpMethod.DELETE.is(request.getMethod())) {
```

Comment:
This kind of logic strikes me as something that should be able to be handled at a more abstract routing level.  Have you had a chance to look for such items?